manifests/database.pp

```diff
@@ -1,9 +1,14 @@
 # Set up the foreman database
 class foreman::database {
   if $::foreman::db_manage {
-    validate_string($::foreman::admin_username, $::foreman::admin_password)
-
     $db_class = "foreman::database::${::foreman::db_type}"
+    class { $db_class:
+      notify => Foreman_config_entry['db_pending_migration'],
```

Comment:
This might be missing if `db_manage` is true, but `db_manage_rake` is false. I'd suggest either making it conditional on both (a separate conditional that chains the two if both parameters are set) or change the rake task block to `if $::foreman::db_manage or $::foreman::db_manage_rake` so db_manage_rake has no function when db_manage is set (I'm unsure if this is desirable).